### ISSUE TITLE
Change TileType to Flooring, and have proper objects for it

### ISSUE THREAD

**Issue by koosemose** [TeamPorcupine/ProjectPorcupine#1522]
Currently TileType is just an Enum, and this causes a need for lots of special cases and odd workarounds to achieve similar behavior to everything else we build. I propose it is changed to Flooring and made a proper Buildable object like pretty much everything else we build. While I suspect this will require a lot of changes, it should also allow future features to be easier to develop, and lead to cleaner code in the end anywhere we currently deal with TileType.

**Comment by NogginBops** [comment]
TileType is not an Enum, it's a full class and it supports xml moddabilty and everything. Or am I missing something.

**Comment by koosemose** [comment]
Sorry, yeah, not an Enum (I keep forgetting that it no longer is, because the most common way to access it mimics it having been an enum), but it still kind of behaves similarly to how it did when it was an enum. It's a thing we build, sort of (we change the TileType), but behaves nothing like other things we build.

To me, at least, it makes more sense to have it something you just place on a tile, rather than a TileType, there is no need for things like TileType.Empty, when you could just have a floor that you build, and if it's empty, it just doesn't have a floor set.

**Comment by BraedonWooding** [comment]
Yes, I support this change.  It makes it sooo much cleaner and nasty enums hehe (for this case).
